src/service/resources-impl.js

```diff
@@ -264,6 +264,12 @@ export class ResourcesImpl {
       }
       // With IntersectionObserver, we only need to handle viewport resize.
       if (this.relayoutAll_ || !this.intersectionObserver_) {
+        // Unfortunately, a viewport size change invalidates all premeasurements.
+        if (this.intersectionObserver_) {
```

Comment:
Nit: I would move this out of the outer conditional, and do: `this.relayoutAll_ && this.intersectionObserver_`

Reply:
done!